### PR TITLE
Add Serverless integration tests for kyma 2.0 on k3s

### PR DESCRIPTION
**Pull request by moelsayed**
**Description**
Use kyma cli to deploy lightweight kyma stack.
Changes proposed in this pull request:

- Use kyma cli to deploy lightweight kyma stack.  

**Related issue(s)** https://github.com/kyma-project/kyma/issues/11257

### CODE REVIEW

File: prow/scripts/cluster-integration/serverless-integration-k3s.sh [hunk 1]
```diff
@@ -95,29 +137,54 @@ install::k3s() {
...
+  # shellcheck disable=SC2004
+  while [[ $(kubectl get nodes -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True" ]]; do echo "Waiting for cluster nodes to be ready, elapsed time: $(( $SECONDS/60 )) min $(( $SECONDS % 60 )) sec"; sleep 2; done
+  host::patch_coredns
+
+  echo "--> Deploying cluster-essentials"
+  kubectl apply -f "$KYMA_SOURCES_DIR/resources/cluster-essentials/files" -n kyma-system
 
-kubectl apply -f "$KYMA_SOURCES_DIR/resources/cluster-essentials/files" -n kyma-system
-helm upgrade --atomic --create-namespace -i serverless "$KYMA_SOURCES_DIR/resources/serverless" -n kyma-system --set "$REGISTRY_VALUES",global.ingress.domainName="$DOMAIN" --wait
+  echo "--> Deploying Serverless"
```
Comment (kwiatekus) [inline prow/scripts/cluster-integration/serverless-integration-k3s.sh:185]: why installing serverless again ? serverless is included in line 165

File: prow/scripts/cluster-integration/serverless-integration-k3s.sh [hunk 2]
```diff
@@ -95,29 +137,54 @@ install::k3s() {
...
+
+  echo "--> Creating k8s cluster via k3s"
+  install::k3s
+
+  # shellcheck disable=SC2004
+  while [[ $(kubectl get nodes -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True" ]]; do echo "Waiting for cluster nodes to be ready, elapsed time: $(( $SECONDS/60 )) min $(( $SECONDS % 60 )) sec"; sleep 2; done
+  host::patch_coredns
+
+  echo "--> Deploying cluster-essentials"
+  kubectl apply -f "$KYMA_SOURCES_DIR/resources/cluster-essentials/files" -n kyma-system
```
Comment (kwiatekus) [inline prow/scripts/cluster-integration/serverless-integration-k3s.sh:183]: why installing cluster essentials manually. already included in line 165

Comment (kwiatekus) [reply prow/scripts/cluster-integration/serverless-integration-k3s.sh:183]: sorry didnt see the if (alpha thingy)

File: prow/scripts/cluster-integration/serverless-integration-k3s.sh (hunk 1 above)
Comment (kwiatekus) [reply prow/scripts/cluster-integration/serverless-integration-k3s.sh:185]: if alpha thingy - surry

**Review by tgorgol — APPROVED** [review]
Great job! LGTM

**Review by tgorgol — DISMISSED** [review]

File: prow/scripts/cluster-integration/serverless-integration-k3s.sh [hunk 3]
```diff
@@ -85,6 +88,45 @@ docker run -d \
   eu.gcr.io/kyma-project/test-infra/docker-registry-2:20200202
 }
 
+install::kyma_cli() {
```
Comment (dekiel) [inline prow/scripts/cluster-integration/serverless-integration-k3s.sh:91]: Can you use existing function for installing kyma_cli? See https://github.com/kyma-project/test-infra/blob/c4d392462597523d632df0c42f8332d0ae52f3d8/prow/scripts/lib/kyma.sh#L114

**Review by dekiel — CHANGES_REQUESTED** [review]

File: prow/scripts/cluster-integration/serverless-integration-k3s.sh (hunk 3 above)
Comment (moelsayed) [reply prow/scripts/cluster-integration/serverless-integration-k3s.sh:91]: Unfortunately, no. This script is [sent over](https://github.com/kyma-project/test-infra/pull/3656/files#diff-7588c996360f463f2a29b2718d12f8986ed77df00fbf089c6a5c71adb805fc64R117) to bash over gcloud-ssh. It has to be self-contained and can't source any external files.

**Review by dekiel — APPROVED** [review]

**Pull request merged**